### PR TITLE
add sig for === for NilClass

### PR DESCRIPTION
**Pull request by maruth-stripe**
### Test plan

See included automated tests.

### CODE REVIEW

File: rbi/core/nil_class.rbi [hunk 1]
```diff
@@ -16,7 +16,13 @@ class NilClass < Object
   # [`Object`](https://docs.ruby-lang.org/en/2.7.0/Object.html), effectively the
   # same as calling `#==`, but typically overridden by descendants to provide
   # meaningful semantics in `case` statements.
-  def ===(_); end
+  sig do
+    params(
+        arg0: BasicObject,
```
Comment (ilyailya) [inline rbi/core/nil_class.rbi:21]: would it make sense to widen the argument's type to `T.anything`?

Comment (jez) [reply rbi/core/nil_class.rbi:21]: +1

**Review by jez — APPROVED** [review]

**Comment by maruth-stripe** [comment]
wait checking why the diff got so large

**Comment by jez** [comment]
> wait checking why the diff got so large

I think you have rubyfmt on somehow.

**Review by ilyailya — APPROVED** [review]

**Pull request merged**